src/FxCop/Desktop.Analyzers/Core/DesktopAnalyzersResources.resx

```diff
@@ -118,48 +118,36 @@
-  </data>
-  <data name="DoNotUseRIPEMD160Description" xml:space="preserve">
-    <value>This type implements RIPEMD160, a cryptographically insecure hashing function. Hash collisions are computationally feasible for the RIPEMD hash algorithms. Replace this usage with a SHA-2 family hash algorithm (SHA512, SHA384, SHA256).</value>
+    <value>What: {0} is catching corrupted state exception.
+
+Why: This could mask errors (such as access violations), resulting in inconsistent state of execution or making it easier for attackers to compromise system.
+
+How: Modify {0} to catch and handle a more specific set of exception type(s) than {1} or re-throw the exception</value>
   </data>
-  <data name="DoNotUseDSA" xml:space="preserve">
-    <value>Do not use insecure cryptographic algorithm DSA.</value>
+  <data name="DoNotUseBrokenCryptographicAlgorithmsMessage" xml:space="preserve">
+    <value>WHAT: {0} uses a broken cryptographic algorithm {1} 
+ 
+WHY: An attack making it computationally feasible to break {1} exists. This allows attackers to break the cryptographic guarantees {1} is designed to provide. Depending on the type and application of this cryptographic algorithm, this may allow attackers to read enciphered messages, tamper with enciphered  messages, forge digital signatures, tamper with hashed content, or otherwise compromise any cryptosystem based on {1}. 
+ 
+HOW: Replace encryption uses with the AES algorithm (AES-256, AES-192 and AES-128 are acceptable) with a key length greater than or equal to 128 bits. Replace hashing uses with a hashing function in the SHA-2 family, such as SHA512, SHA384, or SHA256. Replace digital signature uses with RSA with a key length greater than or equal to 2048-bits, or ECDSA with a key length greater than or equal to 256 bits.</value>
   </data>
-  <data name="DoNotUseDSADescription" xml:space="preserve">
-    <value>This type implements DSA, a cryptographically insecure signature-creation mechanism. Replace this usage with RSA or Diffie-Hellman with a key length greater than or equal to 2048-bits, or ECDSA or ECDH with a key length greater than or equal 256 bits.</value>
+  <data name="DoNotUseBrokenCryptographicAlgorithms" xml:space="preserve">
+    <value>Do Not Use Broken Cryptographic Algorithms</value>
   </data>
-  <data name="DoNotUseTripleDES" xml:space="preserve">
-    <value>Do not use insecure cryptographic algorithm TripleDES.</value>
+  <data name="DoNotUseWeakCryptographicAlgorithms" xml:space="preserve">
+    <value>Do Not Use Weak Cryptographic Algorithms</value>
   </data>
-  <data name="DoNotUseTripleDESDescription" xml:space="preserve">
-    <value>This type implements TripleDES, a cryptographically insecure encryption algorithm. Replace this usage with an AES encryption algorithm (AES-256, AES-192 and AES-128 are acceptable) with a key length greater than or equal to 128 bits.</value>
+  <data name="DoNotUseWeakCryptographicAlgorithmsMessage" xml:space="preserve">
```

Comment:
Same as above:
Message should be "{0} uses a weak cryptographic algorithm {1}."
Description should be "Cryptographic algorithms degrade over time as attacks become for advances to attacker get access to more computation. Depending on the type and application of this cryptographic algorithm, further degradation of it's cryptographic strength may allow attackers to read enciphered messages, tamper with enciphered  messages, forge digital signatures, tamper with hashed content, or otherwise compromise any cryptosystem based on this algorithm. Replace encryption uses with the AES algorithm (AES-256, AES-192 and AES-128 are acceptable) with a key length greater than or equal to 128 bits. Replace hashing uses with a hashing function in the SHA-2 family, such as SHA-2 512, SHA-2 384, or SHA-2 256."